src/ir.cpp

```diff
@@ -10574,6 +10574,9 @@ static IrInstruction *ir_analyze_enum_to_union(IrAnalyze *ira, IrInstruction *so
         result->value.special = ConstValSpecialStatic;
         result->value.type = wanted_type;
         bigint_init_bigint(&result->value.data.x_union.tag, &val->data.x_enum_tag);
+        result->value.data.x_union.payload = create_const_vals(1);
```

Comment:
I don't think this change is valid. `ConstValSpecialStatic` means that the value is initialized and has a meaningful value, but it's not being set here. This would have to be `ConstValSpecialUndef`, but that doesn't seem like what we want here. If we get to this point, it's supposed to be guaranteed that the union field type is `void`, and that means there would be nothing to intialize.

I think the problem is above:
` if (type_has_bits(union_field->type_entry)) {`

It should check for `type_has_one_possible_value`, I think. And then we need to find other code that initializes a value where the type has one possible value (I know it exists somewhere) and turn that into a helper function. Because sometimes you have to initialize the ConstExprValue, e.g. if it's a `u0` you still have to initialize the BigInt to 0.

Reply:
@andrewrk I see. I'll have a look around and see if I can find this code.